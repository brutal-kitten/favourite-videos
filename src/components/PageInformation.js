import React, { Component } from 'react';
import PreviousPage from './PreviousPage';
import NextPage from './NextPage';

class PageInformation extends Component {

  //calls function that recalculate and set state (startIndex, currentPage) in Pagination component
  handleChange = (event) => {

    event.preventDefault();
    let newPage = parseInt(event.target.value, 10);
    let currentPage = (newPage > parseInt(this.props.totalPages, 10) || newPage === 0 || newPage === 1) ? 1 : newPage;
    this.props.setStartIndex(currentPage);
  }


  render () {

    return (
      <div className="pageInfo">
        <PreviousPage
          currentPage={this.props.currentPage}
          setStartIndex={this.props.setStartIndex}
        />
        <div className="pageInput">
          <input id="selectPage" min="1" type="number" max={this.props.totalPages} value={this.props.currentPage} onChange={(event) => this.handleChange(event)} />
        </div>
        <div className="totalPage">of</div>
        <span className="totalPagesNumber">{this.props.totalPages}</span>
        <NextPage
          currentPage={this.props.currentPage}
          totalPages={this.props.totalPages}
          setStartIndex={this.props.setStartIndex}
        />
      </div>
    )
  }
}

export default PageInformation
