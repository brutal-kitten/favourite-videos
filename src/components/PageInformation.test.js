import React from 'react';
import ReactDOM from 'react-dom';
import PageInformation from './PageInformation.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders properly', () => {
  const tree = TestRenderer.create(
    <PageInformation />).toJSON();
  expect(tree).toMatchSnapshot();
});

const mockSetStartIndex = jest.fn();
const props = {
  setStartIndex: mockSetStartIndex,
  totalPages: 3,
  currentPage: 2
}

const wrapper = shallow(<PageInformation {...props}/>);

it('should show current page given from props', () => {
  expect(wrapper.find('div.pageInput')
    .render()
    .html())
    .toBe(`<input type="number" tabindex="0" id="selectPage" min="1" max="3" value="${props.currentPage}">`);
})


describe('when user choose page and there is such page', () => {
  wrapper.find('input').simulate('change', {
    preventDefault: () => {},
    target: {
      value: 3
    },
  });

  it('should call function setStartIndex once', () => {
    expect(mockSetStartIndex.mock.calls.length).toBe(1);
  });

  it('should call function setStartIndex with one argument', () => {
    expect(mockSetStartIndex.mock.calls[0][0]).toBe(3);
  });
});


describe('when user choose page and there is no such page', () => {

  const mockSetStartIndex2 = jest.fn();
  const props2 = {
    setStartIndex: mockSetStartIndex2,
    totalPages: 3,
    currentPage: 2
  }
  const wrapper2 = shallow(<PageInformation {...props2}/>);
  wrapper2.find('input').simulate('change', {
    preventDefault: () => {},
    target: {
      value: (props2.totalPages + 2)
   },
  });

  it('should call function setStartIndex once', () => {
    expect(mockSetStartIndex2.mock.calls.length).toBe(1);
  });

  it('should call function setStartIndex with one argument - "1"', () => {
    expect(mockSetStartIndex2.mock.calls[0][0]).toBe(1);
  });
});
