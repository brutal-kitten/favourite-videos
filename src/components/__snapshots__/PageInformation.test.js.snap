// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="pageInfo"
>
  <div
    className="previousPage"
  >
    <button
      className="buttonPreviousPage"
      onClick={[Function]}
      tabIndex="0"
    >
      <span
        className="glyphicon glyphicon-chevron-left"
      />
    </button>
  </div>
  <div
    className="pageInput"
  >
    <input
      id="selectPage"
      min="1"
      onChange={[Function]}
      tabIndex="0"
      type="number"
    />
  </div>
  <div
    className="totalPage"
  >
    of
  </div>
  <span
    className="totalPagesNumber"
  />
  <div
    className="nextPage"
  >
    <button
      className="buttonNextPage"
      onClick={[Function]}
      tabIndex="0"
    >
      <span
        className="glyphicon glyphicon-chevron-right"
      />
    </button>
  </div>
</div>
`;
